Bump up nodejs version and upgrade dependencies

akurinnoy opened the pull request:
<!-- Please review the following before submitting a PR:
Che's Contributing Guide: https://github.com/eclipse/che/blob/master/CONTRIBUTING.md
Pull Request Policy: https://github.com/eclipse/che/wiki/Development-Workflow#pull-requests

COMMITTERS: please include labels on each PR. Labels are listed here: https://github.com/eclipse/che/wiki/Labels but at a minimum you should include `kind/..` and `Dev Open Pull Request Status` labels.
-->

### What does this PR do?

1. Increases minimal supported version of nodeJS in `package.json`
2. Updates Dockerfiles to use nodeJS v16 images
3. Upgrades dependencies packages within the specified range

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/21237

### Is it tested? How?
<!-- 
Please provide instructions here which scenario you fix/implement
and in which way you tested it, provide as much as you think reviewer
needs to do the same.
-->

olexii4 commented on packages/common/package.json:
```diff
@@ -7,7 +7,7 @@
   "main": "lib/index.js",
   "engines": {
     "yarn": ">=1.20.0",
-    "node": ">=12"
```
Do we really need it?

olexii4 commented on packages/dashboard-frontend/package.json:
```diff
@@ -16,7 +16,7 @@
   "license": "EPL-2.0",
   "engines": {
     "yarn": ">=1.20.0",
-    "node": ">=12"
```
Do we really need it?

akurinnoy commented on package.json:
```diff
@@ -12,7 +12,7 @@
   "homepage": "https://github.com/eclipse-che/che-dashboard",
   "engines": {
     "yarn": ">=1.20.0",
-    "node": ">=12"
```
Agree, I added the fixup

olexii4 reviewed (approved):
Good job!

ibuziuk reviewed (approved):
:+1: looks like one conflict on the rhel dockerfile needs to be resolved

akurinnoy commented:
/retest

akurinnoy commented:
@nickboldt please note that I updated `rhel.Dockerfile`. Since nodeJS v12 is reaching EOL I changed images to `registry.access.redhat.com/ubi8/nodejs-16:1-18`. Will that work for you?

 https://github.com/eclipse-che/che-dashboard/pull/480/files#diff-d62ecfc7db2707ad7af6609a692e3b0237d5b063661a2fac1dd115b2a63ea1c8R11
https://github.com/eclipse-che/che-dashboard/pull/480/files#diff-d62ecfc7db2707ad7af6609a692e3b0237d5b063661a2fac1dd115b2a63ea1c8R36

nickboldt commented on .github/workflows/pr.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     strategy:
       matrix:
-        node-version: [12.x, 14.x]
+        node-version: [14.x, 16.x]
```
if we've updated to build with 16, why also use 14 here?

akurinnoy replied:
Took this from our chatter:
The reason is to minimize switching environments forth and back to different types of versions while working on a few projects and switching between them.

nickboldt commented on package.json:
```diff
@@ -12,7 +12,7 @@
   "engines": {
     "yarn": ">=1.20.0",
-    "node": ">=12"
+    "node": ">=14"
```
why not `>=16`?

akurinnoy replied:
answered https://github.com/eclipse-che/che-dashboard/pull/480#discussion_r833380337

nickboldt reviewed (approved):
generally OK with this, but not sure why wer're doing anything for Node 14 support. Why not just Node 16 minimum?

nickboldt commented:
> build-and-test (12.x) 

Should this PR check also be updated?
